app/controllers/explore.js

```diff
@@ -74,20 +70,32 @@ export default Controller.extend({
       filterOptions.pushObject(this.get('categoryFilter'));
 
       if (subCategory) {
-        this.get('filterOptions').pushObject(this.get('subCategoryFilter'));
+        filterOptions.pushObject(this.get('subCategoryFilter'));
       }
     }
 
     if (eventType) {
-      this.get('filterOptions').pushObject(this.get('typeFilter'));
+      filterOptions.pushObject(this.get('typeFilter'));
     }
 
-    return this.get('store').query('event', {
+    let filteredResults = [];
+
+    await this.get('store').query('event', {
       sort   : 'starts-at',
       filter : filterOptions
+    }).then(results => {
+      filteredResults = results.filter(event => {
```

Comment:
To be done on the server-side.

https://flask-rest-jsonapi.readthedocs.io/en/latest/filtering.html